src/proj.h

```diff
@@ -328,6 +340,17 @@ extern const PJ      *proj_shutdown;
 #endif
 
 
+/* Set or read error level */
+int  proj_errno (PJ *P);
+void proj_errno_set (PJ *P, int err);
```

Comment:
Yes, the proj errno numbers (which are negative) are quite specific (and actually much too specific in my opinion), but `pj_strerrno()` supports the generic POSIX positive errno values as well. Especially I tend to use EINVAL (invalid argument) symbol for quite a lot of generic reporting. While not strictly C89 (I think only EDOM and ERANGE are C89 required), I am yet to see a platform where EINVAL is not defined.

Otherwise, I much agree that fighting for slim APIs is the good fight: Much of the chaos of PROJ.4 is the result of API sprawl. Obviously creeping featurism and clunky design is easier to realise in hindsight, than to prevent in foresight. And care, good taste, real world experience, and a sharp razor are the right tools to ensure that an API stays fresh.  I just think my real world experience differ from yours wrt. error reporting - I have always found it very useful to be able to set an error level on a state object from user/application level code, and to try to make sure that system level code turned into no-ops when handed an error flagged state object, so I would much appreciate to keep `pj_set_errno()` in the API. I would probably be more eager to move `pj_context_errno_set()` to a `proj_plumbing.h` (or whatever) internal system tools file.

Regarding the lack of a proj internal include file: Wasn't the idea of introducing `proj_api.h`to demote `projects.h` to "internal include" status?